event-exporter/sinks/stackdriver/monitored_resource_factory.go

```diff
@@ -0,0 +1,138 @@
+package stackdriver
+
+import (
+	sd "google.golang.org/api/logging/v2"
+
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+type resourceModelVersion string
+
+const (
+	newTypes = resourceModelVersion("new")
+	oldTypes = resourceModelVersion("old")
+)
+
+const (
+	//sd.MonitoredResource Old types:
+	gkeCluster = string("gkeCluster")
+
+	//sd.MonitoredResource New types:
+	k8sCluster = string("k8sCluster")
+	k8sPod     = string("k8sPod")
+	k8sNode    = string("k8sNode")
+
+	//sd.MonitoredResource Labels:=
+	clusterName    = string("clusterName")
+	location       = string("location")
+	projectID      = string("projectID")
+	podName        = string("podName")
+	nodeName       = string("nodeName")
+	namespaceName  = string("namespaceName")
+	involvedObject = string("involvedObject")
+
+	//api_v1.Event InvolvedObject.Kind:
+	pod  = string("Pod")
+	node = string("Node")
+)
+
+type monitoredResourceFactory interface {
+	resourceFromEvent(event *api_v1.Event) *sd.MonitoredResource
+	defaultMonitoredResource() *sd.MonitoredResource
+}
+
+type monitoredResourceFactoryImpl struct {
+	defaultResource *sd.MonitoredResource
+	resourceModel   resourceModelVersion
+	config          *resourceModelConfig
+	labels          map[string]string
+}
+
+func newMonitoredResourceFactory(config *resourceModelConfig) monitoredResourceFactory {
+
+	labels := defaultLabels(config)
+
+	resource := &sd.MonitoredResource{
+		Labels: labels,
+	}
+
+	if config.resourceModel == oldTypes {
+		resource.Type = gkeCluster
+	} else {
+		resource.Type = k8sCluster
+	}
+
+	factory := &monitoredResourceFactoryImpl{
+		defaultResource: resource,
+		resourceModel:   config.resourceModel,
+		config:          config,
+		labels:          labels,
+	}
+
+	return factory
+}
+func defaultLabels(config *resourceModelConfig) map[string]string {
```

Comment:
nit: the name is slightly misleading IMO. Those are not default labels, but the labels common to all monitored resources in this cluster. What about "commonLabels" or "clusterLabels"?